sda/cmd/ingest/ingest.go

```diff
@@ -162,6 +162,100 @@ func main() {
 
 				continue
 			case "ingest":
+				var fileID string
+				status, err := db.GetFileStatus(delivered.CorrelationId)
+				if err != nil && err.Error() != "sql: no rows in result set" {
+					log.Errorf("failed to get status for file, reason: (%s)", err.Error())
+					if err := delivered.Nack(false, true); err != nil {
+						log.Errorf("failed to Nack message, reason: (%s)", err.Error())
+					}
+
+					continue
+				}
+
+				switch status {
+				case "disabled":
+					fileID, err := db.GetFileID(delivered.CorrelationId)
```

Comment:
I guess I'm missing something, but how do we know we get the same correlation id on re-ingestion?

Reply:
Correlation IDs are linked to the uploaded file so they will be the same until the file is assigned to a dataset.